plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/AIAssistantRegistry.java

```diff
@@ -16,17 +16,44 @@
  */
 package org.jkiss.dbeaver.model.ai;
 
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IExtensionRegistry;
+import org.eclipse.core.runtime.Platform;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.utils.CommonUtils;
+
 public class AIAssistantRegistry {
-    private static final AIAssistantRegistry INSTANCE = new AIAssistantRegistry();
 
-    public static AIAssistantRegistry getInstance() {
-        return INSTANCE;
+    private static AIAssistantRegistry instance = null;
+
+    public synchronized static AIAssistantRegistry getInstance() {
+        if (instance == null) {
+            instance = new AIAssistantRegistry(Platform.getExtensionRegistry());
+        }
+        return instance;
     }
 
-    private AIAssistantRegistry() {
+    private final AIAssistantDescriptor customAssistant;
+
+    public AIAssistantRegistry(IExtensionRegistry registry) {
+        AIAssistantDescriptor customAssistantDescriptor = null;
+        IConfigurationElement[] extElements = registry.getConfigurationElementsFor("com.dbeaver.ai.assistant");
+        for (IConfigurationElement ext : extElements) {
```

Comment:
Shouldn't we, like, query an assistant by its identifier?

Reply:
No